cpp/cmake/thirdparty/get_cuco.cmake

```diff
@@ -19,8 +19,8 @@ function(find_and_configure_cuco VERSION)
     if(RAFT_ENABLE_cuco_DEPENDENCY)
```

Comment:
We should just not call `find_and_configure_cuco` when `RAFT_ENABLE_cuco_DEPENDENCY` is enabled, instead of having the switch inside the function itself.